demisto_sdk/commands/pre_commit/pre_commit_command.py

```diff
@@ -164,27 +160,45 @@ def _get_hooks(self, pre_commit_config: dict) -> dict:
 
         return hooks
 
-    def exclude_python2_of_non_supported_hooks(self) -> None:
+    def exclude_hooks_by_version(self) -> None:
         """
-        This function handles the python2 files.
-        Files with python2 run only the hooks that in PYTHON2_SUPPORTED_HOOKS.
+        This function excludes the files that are not supported by the hook, according to the hook min_version property.
         """
-        python2_files = self.python_version_to_files.get(DEFAULT_PYTHON2_VERSION)
-        if not python2_files:
-            return
-
-        logger.info(
-            f"Python {DEFAULT_PYTHON2_VERSION} files running only with the following hooks: {', '.join(PYTHON2_SUPPORTED_HOOKS)}"
-        )
+        for hook in self.hooks.values():
+            min_version = Hook.get_property(hook["hook"], self.mode, "min_py_version")
+            if not min_version:
+                continue
+            files_to_exclude: Set[Path] = set()
+
+            for version, paths in self.python_version_to_files.items():
+                if Version(version) < Version(min_version):
+                    files_to_exclude.update(path for path in paths)
+            if files_to_exclude:
+                join_files_string = join_files(files_to_exclude)
+                if hook["hook"].get("exclude"):
+                    hook["hook"]["exclude"] += f"|{join_files_string}"
+                else:
+                    hook["hook"]["exclude"] = join_files_string
 
-        join_files_string = join_files(python2_files)
+    def exclude_hooks_by_support_level(self) -> None:
+        """This function excludes the hooks that are not supported by the support level of the file."""
         for hook in self.hooks.values():
-            if hook["hook"]["id"] in PYTHON2_SUPPORTED_HOOKS:
+            support_levels = Hook.get_property(
+                hook["hook"], self.mode, "exclude_support_level"
+            )
+            if not support_levels:
                 continue
-            elif hook["hook"].get("exclude"):
-                hook["hook"]["exclude"] += f"|{join_files_string}"
-            else:
-                hook["hook"]["exclude"] = join_files_string
+            files_to_exclude: Set[Path] = set()
+            for support_level in support_levels:
+                files_to_exclude.update(
+                    path for path in self.support_level_to_files[support_level]
+                )
+            if files_to_exclude:
```

Comment:
this logic should be done inside the hooks class themselves. We're defeating the purpose of the object orientation by doing this way. Put this sort of logic in base hook

Reply:
fixed